### PR TITLE
Dala 4864 filter and pagination on admin requests page

### PR DESCRIPTION
**Pull request by dennis-dfine**
# Pull Request \<Title>
Please also functional test on clone
## Things to do during Peer Review
Please check all boxes before the Pull Request is merged. In case you skip a box, describe in the PRs description (that means: here) why the check is skipped.
- [x] The Github Actions (including Sonarqube Gateway and Lint Checks) are green. This is enforced by Github. 
- [x] A peer-review has been executed
  - [x] The code has been manually inspected by someone who did not implement the feature
  - [x] If this PR includes work on the frontend, at least one `@ts-nocheck` is removed. Additionally, there should not be any `@ts-nocheck` in files modified by this PR. If no `@ts-nocheck` are left: Celebrate :tada: :confetti_ball: type-safety and remove this entry. 
- [x] The PR actually implements what is described in the JIRA-Issue
- [x] At least one test exists testing the new feature
  - [x] If you have created new test files, make sure that they are included in a test container and actually run in the CI
- [x] At least 3 consecutive CI runs are successfully executed to ensure that there are no flaky tests.
- [x] Documentation is updated as required
- [x] The automated deployment is updated if required
- [ ] If there was a database entity class added, there must also be a migration script for creating the corresponding database if flyway is already used by the service
- [ ] IF there are changes done to the framework data models or to a database entity class, the following steps were completed in order
  - [ ] A fresh clone of dataland.com is generated (see Wiki page on "OTC" for details)
  - [ ] The feature branch is deployed to clone with `Reset non-user related Docker Volumes & Re-populate` turned off
  - [ ] It's verified that the CD run is green
  - [ ] The new feature is manually used/tested/observed on the clone server. Testing of the new feature should (also) be done by a second, independent reviewer from the dev team
  - [ ] The feature branch is deployed to dev1 with `Reset non-user related Docker Volumes & Re-populate` turned on, and it's verified that the CD run is green  
- [x] ELSE, the new version is deployed to the dev server "dev1" using this branch
  - [x] Run with setting `Reset non-user related Docker Volumes & Re-populate` turned on 
  - [x] It's verified that this version actually is the one deployed (check gitinfo for branch name and commit id!)
  - [x] It's verified that the CD run is green
  - [x] The new feature is manually used/tested/observed on dev server. Testing of the new feature should (also) be done by a second, independent reviewer from the dev team

### CODE REVIEW

File: dataland-frontend/src/components/pages/AdminAllRequestsOverview.vue [hunk 1]
```diff
@@ -322,19 +334,22 @@ export default defineComponent({
...
      * @param event DataTablePageEvent
      */
     onPage(event: DataTablePageEvent) {
       window.scrollTo(0, 0);
-      if (event.page != this.currentPage) {
-        this.currentPage = event.page;
+      if (event.page != this.currentChunkIndex) {
+        this.currentChunkIndex = event.page;
+        this.firstRowIndex = this.currentChunkIndex * this.rowsPerPage;
+        console.log(event.first);
```
Comment (stewal19) [inline dataland-frontend/src/components/pages/AdminAllRequestsOverview.vue:352]: Why this console.log?

File: dataland-frontend/src/components/pages/AdminAllRequestsOverview.vue [hunk 2]
```diff
@@ -322,19 +334,22 @@ export default defineComponent({
...
+     * Updates the current Page
      * @param event DataTablePageEvent
      */
     onPage(event: DataTablePageEvent) {
       window.scrollTo(0, 0);
-      if (event.page != this.currentPage) {
-        this.currentPage = event.page;
+      if (event.page != this.currentChunkIndex) {
+        this.currentChunkIndex = event.page;
+        this.firstRowIndex = this.currentChunkIndex * this.rowsPerPage;
```
Comment (stewal19) [inline dataland-frontend/src/components/pages/AdminAllRequestsOverview.vue:351]: Does this mean if i go to the second chunk (chunkIndex = 2) that the firstRowIndex is = 200?

File: dataland-frontend/src/components/pages/AdminAllRequestsOverview.vue [hunk 3]
```diff
@@ -232,15 +235,14 @@ export default defineComponent({
     this.availableFrameworks = retrieveAvailableFrameworks();
     this.availableRequestStatus = retrieveAvailableRequestStatus();
     this.getAllRequestsForFilters().catch((error) => console.error(error));
-    this.resetFilterAndSearchBar();
```
Comment (stewal19) [inline dataland-frontend/src/components/pages/AdminAllRequestsOverview.vue:235]: Is is deleted because it was not necessary on mount, correct?

Comment (dennis-dfine) [reply dataland-frontend/src/components/pages/AdminAllRequestsOverview.vue:235]: exactly. on mount the selectionArrays are empty, so no need to reset the filter (which would and did so before, on mount, lead to three api-calls, first due to mount, and two consecutive ones due to resetting the filter).

Comment (stewal19) [reply dataland-frontend/src/components/pages/AdminAllRequestsOverview.vue:235]: ok

File: dataland-frontend/src/components/pages/AdminAllRequestsOverview.vue (hunk 1 above)
Comment (stewal19) [reply dataland-frontend/src/components/pages/AdminAllRequestsOverview.vue]: :)

File: dataland-frontend/tests/component/components/pages/AdminAllRequestsOverview.cy.ts [hunk 4]
```diff
@@ -200,6 +234,38 @@ describe('Component test for the admin-requests-overview page', () => {
     assertEmailAddressExistsInSearchResults(mailDelta);
   }
 
+  /**
+   * Validates onPage Event
+   */
+  function validateOnPageEvent(): void {
+    const secondPageMockResponse = mockRequestsLarge.slice(100);
```
Comment (stewal19) [inline dataland-frontend/tests/component/components/pages/AdminAllRequestsOverview.cy.ts:241]:
```suggestion
    const secondPageMockResponse = mockRequestsLarge.slice(max_rows);
```

**Pull request merged**